validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -301,23 +301,28 @@ private Metadata readMetadataWithValidation(boolean validate) {
       return null;
     }
 
+    JsonNode mergedMetadata = getMergedMetadata(instance);
```

Comment:
Add unit tests for this function please. It might need to be broken up into pieces?

Reply:
Added class LocalDeviceTest with coverage for the two major caes